packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -290,13 +298,13 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
       Annotations.of(this).addError(`Cluster requires at least 2 subnets, got ${subnetIds.length}`);
     }
 
-    this.subnetGroup = new CfnDBSubnetGroup(this, 'Subnets', {
-      dbSubnetGroupDescription: `Subnets for ${id} database`,
-      subnetIds,
+    const subnetGroup = props.subnetGroup ?? new SubnetGroup(this, 'Subnets', {
+      description: `Subnets for ${id} database`,
+      vpc: props.instanceProps.vpc,
+      vpcSubnets: props.instanceProps.vpcSubnets,
+      removalPolicy: props.removalPolicy === RemovalPolicy.RETAIN ? props.removalPolicy : undefined,
     });
-    if (props.removalPolicy === RemovalPolicy.RETAIN) {
-      this.subnetGroup.applyRemovalPolicy(RemovalPolicy.RETAIN);
-    }
```

Comment:
I believe this is a pretty serious change in behavior, no?

Previously, the SubnetGroup had the `Retain` delete policy only if the Cluster had `Retain` - otherwise, it would have `Remove`. But now, it always has `Retain`, as, according to the docs, the default in `SubnetGroup` is `Retain` as well - unless the docs are incorrect for it...?